### PR TITLE
test: migrate login password flow to hono

### PR DESCRIPTION
**Pull request by danielgent**

### CODE REVIEW

File: src/bun.ts [hunk 1]
```diff
@@ -21,7 +20,6 @@ const server = {
 
     return app.fetch(request, {
       ...process.env,
-      oauth2ClientFactory: { create: oAuth2ClientFactory },
```
Comment (danielgent) [inline src/bun.ts:24]: see below

File: test/integration/helpers/test-client.ts [hunk 2]
```diff
@@ -13,8 +13,10 @@ import {
 } from "../../../src/types";
 import { EmailAdapter } from "../../../src/adapters/interfaces/Email";
 import type { Email } from "../../../src/types/Email";
+import { Env } from "../../../src/types";
+import { oAuth2ClientFactory } from "../../../src/services/oauth2-client";
 
-export async function getEnv() {
+export async function getEnv(): Promise<Env & { db: any }> {
```
Comment (danielgent) [inline test/integration/helpers/test-client.ts:19]: :eyes: I've wanted to do this. It keeps tripping me up that we're missing things from the `Env` 

Some of these mocks might not be needed, but I figure that's better than having to work out which we need on a per-test basis

File: test/integration/helpers/test-client.ts [hunk 3]
```diff
@@ -205,6 +205,10 @@ export async function getEnv() {
     data: {
       ...data,
       email: emailAdapter,
+      templates: {
+        get: async (...inputs: any[]) =>
+          `<div>${JSON.stringify(inputs, null, 2)}</div>`,
+      },
```
Comment (danielgent) [inline test/integration/helpers/test-client.ts:211]: copied from wrangler tests

Comment (markusahlstrand) [reply test/integration/helpers/test-client.ts:211]: I'm guessing this basically will add any json into the template which we probably are testing for?

**Review by markusahlstrand — APPROVED** [review]
LGTM

File: src/routes/tsoa/login.ts [hunk 4]
```diff
@@ -169,12 +169,14 @@ export class LoginController extends Controller {
...
-      user_id: params.username,
-    });
+    // this should be commented out right? what is the plan here...
+    // blows up with "FK constraint failed"
+    // await env.data.logs.create({
+    //   category: "login",
+    //   message: "Create authentication code",
+    //   tenant_id: client.tenant_id,
+    //   user_id: params.username,
+    // });
```
Comment (markusahlstrand) [inline src/routes/tsoa/login.ts:179]: Remove commented code
```suggestion
```

**Pull request merged**